core/src/main/java/com/graphhopper/util/PathMerger.java

```diff
@@ -99,6 +99,8 @@ public void doWork(PathWrapper altRsp, List<Path> paths, Translation tr) {
                     if (pathIndex + 1 < paths.size()) {
                         ViaInstruction newInstr = new ViaInstruction(fullInstructions.get(fullInstructions.size() - 1));
                         newInstr.setViaCount(pathIndex + 1);
+                        // Similar to the FinishInstruction, ViaInstructions have lenght=0
```

Comment:
I do not fully understand it. The ViaInstruction and FinishInstruction both should have one point, why is setting to an empty list necessary? And why do we still need this viaPosition/viaCount workaround?